src/GraphQL.MemoryCache/MemoryDocumentCache.cs

```diff
@@ -59,7 +60,7 @@ protected MemoryDocumentCache(IMemoryCache memoryCache, bool disposeMemoryCache,
     /// </summary>
     protected virtual MemoryCacheEntryOptions GetMemoryCacheEntryOptions(ExecutionOptions options)
     {
-        return new MemoryCacheEntryOptions { SlidingExpiration = _options.SlidingExpiration, Size = options.Query!.Length };
+        return new MemoryCacheEntryOptions { SlidingExpiration = _options.SlidingExpiration, Size = options.Query?.Length ?? options.DocumentId?.Length ?? 100 };
```

Comment:
Won't setting 100 as a default be a breaching change?

Reply:
No as Query should be set for any existing code, and if it wasn’t then the code would break. Secondly, either the query or document id should always exist, or there would be nothing to cache against.

However the number is supposed to be representative of the amount of memory required in the cache. The length of the hash isn’t representative. I should write an ast visitor to compute the amount of memory required.

Reply:
Well, unfortunately `GetMemoryCacheEntryOptions` does not pass the `GraphQLDocument` in, so that will have to be a v9 change.  Once it does, we can just grab the `Source.Length` from the document and use that.  So for now, all we have is the document id length to measure.

Reply:
Added overload and marked old one as obsolete.  No breaking change, as if the Query property exists (as it must or the code would have thrown an exception before), it will use the old method.